pkg/arvo/sys/arvo.hoon

```diff
@@ -417,29 +417,67 @@
 ++  de
   =|  fat=(axal)
   |@
+  ++  del
+    |=  pax=path
+    ?~  pax  [~ dir.fat]
+    =/  kid  (~(get by dir.fat) i.pax)
+    ?~  kid  fat
+    $(fat u.kid, pax t.pax)
```

Comment:
It looks like this is (almost) just `=+  (~(dip de fat) pax)  -(fil ~)`. Doesn't this need to rebuild the `axal` on the way up the stack?

Reply:
Yeah, this looks like it will produce just the child axal, not the mutated root axal.

Replacing the last line with this should produce the mutated root axal:

```hoon
[fil.fat (~(put by dir.fat) i.pax $(fat u.kid, pax t.pax))]
```

Reply:
Actually, I think this line, taken from +lop, would allocate less:

```hoon
fat(dir (~(put by dir.fat) i.pax $(fat u.kid, pax t.pax)))
```